File: 2477-minimum-fuel-cost-to-report-to-the-capital/2477-minimum-fuel-cost-to-report-to-the-capital.cpp
class Solution {
public:
    long long res=0,sts;
    long long dfs(int st,int prev,vector<vector<int>>&g,int people=1){
        for(int &i:g[st]){
            if(i==prev)continue;
            people+=dfs(i,st,g);
        }
        if(st!=0)res+=(people+sts-1)/sts;
        return people;
    }
    long long minimumFuelCost(vector<vector<int>>& roads, int seats) {
        sts=seats;
        vector<vector<int>>g(roads.size()+1);
        for(vector<int>&i:roads){
            g[i[0]].push_back(i[1]);
            g[i[1]].push_back(i[0]);
        }
        dfs(0,0,g);
        return res;
    }
};

File: 2477-minimum-fuel-cost-to-report-to-the-capital/2477-minimum-fuel-cost-to-report-to-the-capital.java
class Solution {
    long res=0;int sts;
    public long dfs(int st,int prev,List<List<Integer>>g){
        int p=1;
        for(int i:g.get(st)){
            if(i==prev)continue;
            p+=dfs(i,st,g);
        }
        if(st!=0)res+=(p+sts-1)/sts;
        return p;
    }
    public long minimumFuelCost(int[][] roads, int seats) {
        sts=seats;
        List<List<Integer>>g=new ArrayList();
        for(int i=0;i<=roads.length;i++)g.add(new ArrayList());
        for(int[]r: roads){
            g.get(r[0]).add(r[1]);
            g.get(r[1]).add(r[0]);
        }
        dfs(0,0,g);
        return res;
    }
}